cpp/src/arrow/record_batch.h

```diff
@@ -234,6 +234,53 @@ class ARROW_EXPORT RecordBatchReader {
     return batch;
   }
 
+  class RecordBatchReaderIterator {
+   public:
```

Comment:
It would be nicer to satisfy the [input iterator requirement](https://en.cppreference.com/w/cpp/named_req/InputIterator), i.e. add type definitions (`value_type`, `reference`) as well as equality and post-increment operators.

Reply:
Pardon the lack of rigor, this was a direct translation of the nested class at https://github.com/apache/arrow/blob/31a07be1d9dc2f7c9720cc0fdcd7f083d947aba1/cpp/src/arrow/util/iterator.h#L134 to RecordBatchReader. I'll be sure to add these definitions.